src/Foreign/Convertible.agda

```diff
@@ -1,9 +1,8 @@
 {-# OPTIONS --overlapping-instances #-}
 module Foreign.Convertible where
 
-open import Ledger.Prelude
```

Comment:
I was thinking that we should go from more distant dependencies to closer ones, e.g.
```agda
-- built-in
import Agda.Builtin.*
-- stdlib
import Data.*
import Relation.*
-- stdlib-meta
import Prelude
import Interface.*
-- ledger
import Ledger.Prelude
....
```
In other words, apply the "prelude-at-the-top" rule only inside a dependency group, and order groups as above. WDYT?

Reply:
I like this idea, though I have no preference between ascending or descending distance ordering.  My real preference is to simply use alphabetical order, but @WhatisRT vetoed that idea.  Your suggestion is at least better than haphazard or random ordering.  I just want to know where to look in a long list of imports to find something quickly, but for me the current ordering of imports makes that hard.  I suspect Andre has his reasons for the current orderings of imports and if I bet if I understood those reasons it would be much easier for me to quickly find things.

Reply:
I like this idea to an extent. I think it's good to group similar imports together but the problem is that imports might be similar but from different libraries. Or the other way round: I think `Ledger.Prelude` should be separated from all the other `Ledger` imports.

Since we all have different preferences and you don't want this PR to be stuck in limbo, how about you make in issue & revert the change for now?

Reply:
It seems like a waste of time to revert these to their original arbitrary order, we can still make an issue to change the order to something better without reverting the changes?

Reply:
Yeah, we could, but I think the order you've put them in is arbitrary 😅 It's not alphabetic, not in your proposed order (everything except `Ledger.Prelude` is from the stdlib), and not grouped by theme (which it was originally).

Reply:
For this file indeed I should put `Ledger.Prelude` last to conform to my own convention.

NB: As with all styling directives, I do not propose we enforce my convention strictly 100%, in this case I would prefer the `Foreign.` imports to come last as this is the most "specific" aspect of the Haskell extraction part of the codebase.

However, we are discussing something more general here that should apply to *all* files, and I have more or less "cleaned up" imports everywhere. So, open a separate issue but keep my cleaned up versions without reverting?